--- sw/airborne/gps.c	2010-08-31 17:34:05.838376327 +0200
+++ sw/airborne/gps_new.c	2010-08-31 17:31:18.097499266 +0200
@@ -31,7 +31,9 @@
 
 #include "gps.h"
 #include "latlong.h"
+#ifndef FMS_PERIODIC_FREQ
 #include "sys_time.h"
+#endif
 #include "generated/airframe.h"
 #include "generated/periodic.h"
 
@@ -46,8 +48,15 @@
 #endif
 
 #define DOWNLINK_DEVICE DOWNLINK_GPS_DEVICE
+
+#ifdef FMS_PERIODIC_FREQ
+#include "messages2.h"
+#include "beth/overo_gcs_com.h"
+#define DefaultChannel gcs_com.udp_transport
+#else
 #include "messages.h"
 #include "downlink.h"
+#endif
 
 uint16_t last_gps_msg_t;	/** cputime of the last gps message */
 bool_t gps_verbose_downlink;
